core/lib/zksync_core/src/l1_gas_price/gas_adjuster/mod.rs

```diff
@@ -130,7 +129,7 @@ impl<E: EthInterface> L1GasPriceProvider for GasAdjuster<E> {
 
     fn estimate_effective_pubdata_price(&self) -> u64 {
         // For now, pubdata is only sent via calldata, so its price is pegged to the L1 gas price.
-        self.estimate_effective_gas_price() * L1_GAS_PER_PUBDATA_BYTE as u64
+        self.estimate_effective_gas_price() * self.config.l1_gas_per_pubdata_byte
```

Comment:
The `L1_GAS_PER_PUBDATA_BYTE` variable is used in many other parts of the code, so we risk inconsistency (if config value changes, this component will consider a different value than others). Additionally, it doesn't feel right that in config when we set the mode (rollup/validium) we also need to set this gas per pubdata byte to zero. 


What if we have a trait `GasAdjuster` with all its logic identical to this file, and two structs that implement it:
```
struct RollupGasAdjuster {
    l1_gas_per_pubdata_byte: L1_GAS_PER_PUBDATA_BYTE
}

struct ValidiumGasAdjuster {
    l1_gas_per_pubdata_byte: 0
}

```

This also allows for different DA modes to have slightly different behaviour when predicting L1/DA cost of a transaction